apps/dashboard/src/app/(app)/team/[team_slug]/[project_slug]/insight/webhooks/components/_hooks/useAbiProcessing.ts

```diff
@@ -0,0 +1,132 @@
+"use client";
+
+import { useQueries } from "@tanstack/react-query";
+import { useMemo } from "react";
+import { defineChain } from "thirdweb/chains";
+import { getContract, resolveAbiFromContractApi } from "thirdweb/contract";
+import type { ThirdwebClient } from "thirdweb/dist/types/client/client";
+import type {
+  AbiData,
+  EventSignature,
+  FunctionSignature,
+} from "../_utils/webhook-types";
+
+export function useAbiBatchProcessing({
+  isOpen,
+  thirdwebClient,
+  chainIds,
+  addresses,
+  extractSignatures,
+  type,
+}: {
+  isOpen: boolean;
+  thirdwebClient: ThirdwebClient;
+  chainIds: string[];
+  addresses: string;
+  extractSignatures: (
+    abis: AbiData[],
+  ) => EventSignature[] | FunctionSignature[];
+  type: "event" | "transaction";
+}) {
+  const pairs = useMemo(() => {
+    const result: { chainId: string; address: string }[] = [];
+    for (const chainId of chainIds) {
+      for (const address of (addresses || "")
+        .split(",")
+        .map((a) => a.trim())
+        .filter(Boolean)) {
+        result.push({ chainId, address });
+      }
+    }
+    return result;
+  }, [chainIds, addresses]);
+
+  const queries = useMemo(() => {
+    return pairs.map(({ chainId, address }) => ({
+      queryKey: ["abi", chainId, address, type],
+      queryFn: async () => {
+        // eslint-disable-next-line no-restricted-syntax
+        const chainObj = defineChain(Number(chainId));
+        const contract = getContract({
+          client: thirdwebClient,
+          address,
+          chain: chainObj,
+        });
+        return resolveAbiFromContractApi(contract);
+      },
+      enabled: isOpen && !!addresses.trim() && chainIds.length > 0,
+      staleTime: Number.POSITIVE_INFINITY,
+    }));
+  }, [pairs, isOpen, addresses, chainIds, type, thirdwebClient]);
+
+  const results = useQueries({ queries });
```

Comment:
Avoid using `useQueries` - If the number of queries may be large it will crash the page due to a large amount of memory used. If possible, please do all request in a single `useQuery`

Reply:
hm this does not make sense to me, one useQuery would be LESS efficient in this case no?